tests/interlocked_linux_ut/mock_interlocked.c

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdatomic.h>
+#include "mock_interlocked.h"
+
+#undef atomic_fetch_add
+#define atomic_fetch_add mock_atomic_fetch_add
+#define mock_atomic_fetch_add(X, Y) _Generic((X), \
+    volatile_atomic int16_t*: mock_atomic_fetch_add_16, \
+    volatile_atomic int32_t*: mock_atomic_fetch_add_32, \
+    volatile_atomic int64_t*: mock_atomic_fetch_add_64 \
+)(X, Y)
```

Comment:
Yay for getting C generics working for this mocking!

But this reminds me that *maybe* we should make sure this library works without C11 on Linux.

I think that the cleanest way is to just remove all of the support for this in the CMakeLists.txt in that case. Adding a comment there...